diff --git lib/sensu/settings/validators/check.rb lib/sensu/settings/validators/check.rb
index 1348e4a..de325c4 100644
--- lib/sensu/settings/validators/check.rb
+++ lib/sensu/settings/validators/check.rb
@@ -61,7 +61,7 @@ module Sensu
             invalid(check, "check name must be a string")
           must_match_regex(/^[\w\.-]+$/, check[:name]) ||
             invalid(check, "check name cannot contain spaces or special characters")
-          must_be_a_string(check[:command]) ||
+          (must_be_a_string(check[:command]) or must_be_a_string(check[:extension])) ||
             invalid(check, "check command must be a string")
           must_be_a_numeric_if_set(check[:timeout]) ||
             invalid(check, "check timeout must be numeric")
