--- a/lib/sensu/server.rb	2014-04-19 11:14:58.000000000 -0500
+++ b/lib/sensu/server.rb	2014-07-02 14:51:31.659610191 -0500
@@ -471,6 +471,7 @@
                     :status => check[:status],
                     :issued => check[:issued],
                     :check_parameters => {
+                      :interval => check[:interval] ? check[:interval] : nil,
                       :refresh => check[:refresh] ? check[:refresh] : nil,
                       :occurrences => check[:occurrences] ? check[:occurrences] : nil,
                       :dependencies => check[:dependencies] ? check[:dependencies] : nil
