diff --git a/lib/sensu/client.rb b/lib/sensu/client.rb
index bb09cc2..623942e 100644
--- lib/sensu/client.rb
+++ lib/sensu/client.rb
@@ -126,8 +126,8 @@ module Sensu
         :check => check
       })
       check[:executed] = Time.now.to_i
-      extension = @extensions[:checks][check[:name]]
-      extension.safe_run do |output, status|
+      extension = @extensions[:checks][check[:extension]]
+      extension.safe_run(check) do |output, status|
         check[:output] = output
         check[:status] = status
         publish_result(check)
@@ -151,9 +151,9 @@ module Sensu
         else
           execute_check_command(check)
         end
-      else
-        if @extensions.check_exists?(check[:name])
-          run_check_extension(check)
+        elsif check.has_key?(:extension)
+          if @extensions.check_exists?(check[:extension])
+           run_check_extension(check)
         else
           @logger.warn('unknown check extension', {
             :check => check
