diff --git a/lib/sensu/server.rb b/lib/sensu/server.rb
index 5d3ed9e..7da0d45 100644
--- lib/sensu/server.rb
+++ lib/sensu/server.rb
@@ -418,6 +418,12 @@ module Sensu
                     event[:occurrences] = previous_occurrence[:occurrences] + 1
                   end
                   event[:action] = is_flapping ? :flapping : :create
+                  event[:check_paramters] = {
+                    :interval => check[:interval] ? check[:interval] : nil,
+                    :refresh => check[:refresh] ? check[:refresh] : nil,
+                    :occurrences => check[:occurrences] ? check[:occurrences] : nil,
+                    :dependencies => check[:dependencies] ? check[:dependencies] : nil
+                  }
                   @redis.hset('events:' + client[:name], check[:name], MultiJson.dump(event)) do
                     unless check[:handle] == false
                       handle_event(event)
