--- lib/sensu/server.rb  2014-10-24 12:59:23.188145592 -0500
+++ lib/sensu/server.rb  2014-10-24 13:15:08.818064942 -0500
@@ -453,11 +453,18 @@
     def setup_results
       @logger.debug('subscribing to results')
       @transport.subscribe(:direct, 'results', 'results', :ack => true) do |message_info, message|
-        result = MultiJson.load(message)
-        @logger.debug('received result', {
-          :result => result
-        })
-        process_result(result)
+        begin
+          result = MultiJson.load(message)
+          @logger.debug('received result', {
+            :result => result
+          })
+          process_result(result)
+        rescue MultiJson::ParseError => error
+          @logger.error('failed to parse result payload', {
+            :message => message,
+            :error => error.to_s
+          })
+        end
         EM::next_tick do
           @transport.ack(message_info)
         end
